torch/_inductor/ir.py

```diff
@@ -8012,7 +8012,10 @@ def add_submodule(self, block, prefix):
         return name
 
     def add_indirect(self, size):
-        var = sympy_index_symbol_with_prefix(SymT.INDIRECT, len(self.indirect_vars))
+        from torch.utils._sympy.symbol import make_symbol
+
+        # Note: indirect index variables can be negative, positive or 0.
+        var = make_symbol(SymT.INDIRECT, len(self.indirect_vars), integer=True)
```

Comment:
Ah, right, we do the wrapping within `indirect_indexing` in codegen, so I guess the positive condition is correct here.

Reply:
That makes sense, I'll change that back because we know it must be non-negative after wrapping it.

> wrapping of negative values should happen before the real value is assigned to the symbolic value
Btw I'm trying to understand what this looks like. Can you help me understand? Thanks!

Reply:
see the codegen for indirect_index in common.py